include/seastar/http/reply.hh

```diff
@@ -111,6 +111,38 @@ struct reply {
         network_connect_timeout = 599, //!< network_connect_timeout
     } _status;
 
+    /**
+     * HTTP status classes
+     * See https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
+     *
+     * 1xx: Informational - Request received, continuing process
+     * 2xx: Success - The action was successfully received, understood, and accepted
+     * 3xx: Redirection - Further action must be taken in order to complete the request
+     * 4xx: Client Error - The request contains bad syntax or cannot be fulfilled
+     * 5xx: Server Error - The server failed to fulfill an apparently valid request
+     */
+    enum class status_class : uint8_t {
+        informational=1,
+        success=2,
+        redirection=3,
+        client_error=4,
+        server_error=5,
+        unclassified
+    };
+
+    /**
+     * Classify http status
+     * @param http_status the http status \ref status_type
+     * @return one of the \ref status_class values
+     */
+    static constexpr status_class classify_status(status_type http_status) {
+        auto sc = static_cast<uint16_t>(http_status) / 100;
```

Comment:
nit, the underlying type of `status_type` is `int`. as by default if the underlying type is not specified, `enum class status_type` uses `int` as its underlying type. so IMHO, `uint16_t` is a, well, "magic type". if we want to perform integer math on it, i think, a better way is probably to cast it to `static_cast<std::underlying_type_t<status_type>>(http_status)`. BTW, in C++23, we would be able to use `std::to_underlying(http_status)`. but we are not there yet..

Reply:
yup, looks cleaner this way